Wellthier/API/ExerciseAPIManager.m

```diff
@@ -57,23 +51,38 @@ - (NSMutableURLRequest *) urlRequestForItems: (NSString *) itemsName {
     return request;
 }
 
-- (void) fetchBodyParts: (void(^)(NSArray *bodyParts, NSError *error))completion {
-    NSMutableURLRequest *request = [self urlRequestForItems:@"/bodyPartList"];
-    
+- (void) fetchExerciseByID: (NSString *) exerciseID withCompletion: (void(^)(Exercise *exercise, NSError *error))completion {
+    NSMutableURLRequest *request = [self urlRequestForItems:[NSString stringWithFormat:@"/exercise/%@", exerciseID]];
+    [self makeRequest:request completion:^(id dataDictionary, NSError *error) {
+                                                            if (error) {
+                                                            } else {
+                                                                if ([dataDictionary isKindOfClass:[NSDictionary class]]) {
+                                                                    Exercise *exercise = [Exercise new];
```

Comment:
You can make a initialize function for Exercise. Something like `newWithName: bodyPart: equipment: ...'